src/test/java/com/commercetools/sync/commons/utils/ResourceCustomUpdateActionUtilsTest.java

```diff
@@ -0,0 +1,569 @@
+        final Map<String, JsonNode> oldCustomFields = new HashMap<>();
+        oldCustomFields.put("backgroundColor", JsonNodeFactory.instance.objectNode().put("de", "rot").put("en", "red"));
+
+        final Map<String, JsonNode> newCustomFields = new HashMap<>();
+        newCustomFields.put("invisibleInShop", JsonNodeFactory.instance.booleanNode(true));
+
+        final List<UpdateAction<Category>> customFieldsActions =
+            buildNewOrModifiedCustomFieldsUpdateActions(oldCustomFields, newCustomFields, mock(Category.class),
+                new CategoryCustomActionBuilder(),null, category -> null);
+
+        assertThat(customFieldsActions).isNotNull();
+        assertThat(customFieldsActions).isNotEmpty();
+        assertThat(customFieldsActions).hasSize(1);
+        assertThat(customFieldsActions.get(0)).isInstanceOf(SetCustomField.class);
+    }
+
+
+    @Test
+    public void buildNewOrModifiedCustomFieldsUpdateActions_WithNoNewOrModifiedCustomFields_ShouldNotBuildActions() {
+        final Map<String, JsonNode> oldCustomFields = new HashMap<>();
+        oldCustomFields.put("invisibleInShop", JsonNodeFactory.instance.booleanNode(true));
+        oldCustomFields.put("backgroundColor", JsonNodeFactory.instance.objectNode().put("de", "rot").put("en", "red"));
+
+        final Map<String, JsonNode> newCustomFields = new HashMap<>();
+        newCustomFields.put("invisibleInShop", JsonNodeFactory.instance.booleanNode(true));
+
+        final List<UpdateAction<Category>> customFieldsActions =
+            buildNewOrModifiedCustomFieldsUpdateActions(oldCustomFields, newCustomFields, mock(Category.class),
+                new CategoryCustomActionBuilder(), null, categoryResource -> null);
+
+        assertThat(customFieldsActions).isNotNull();
+        assertThat(customFieldsActions).isEmpty();
+    }
+
+    @Test
+    public void buildRemovedCustomFieldsUpdateActions_WithRemovedCustomField_ShouldBuildUpdateActions() {
+        final Map<String, JsonNode> oldCustomFields = new HashMap<>();
+        oldCustomFields.put("invisibleInShop", JsonNodeFactory.instance.booleanNode(true));
+        oldCustomFields.put("backgroundColor", JsonNodeFactory.instance.objectNode().put("de", "rot").put("en", "red"));
+
+        final Map<String, JsonNode> newCustomFields = new HashMap<>();
+        newCustomFields.put("invisibleInShop", JsonNodeFactory.instance.booleanNode(true));
+
+        final List<UpdateAction<Category>> customFieldsActions =
+            buildRemovedCustomFieldsUpdateActions(oldCustomFields, newCustomFields, mock(Category.class),
+                new CategoryCustomActionBuilder(), null, category -> null);
+
+        assertThat(customFieldsActions).isNotNull();
+        assertThat(customFieldsActions).isNotEmpty();
+        assertThat(customFieldsActions).hasSize(1);
+        assertThat(customFieldsActions.get(0)).isInstanceOf(SetCustomField.class);
```

Comment:
check if it is a removeaction

Reply:
Please refer to comment: https://github.com/commercetools/commercetools-sync-java/pull/253#discussion_r170023849